gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java

```diff
@@ -96,6 +99,36 @@
     private Map<String, Object> manifestExtensions = Maps.newHashMap();
     private File manifestFile;
 
+    public CreateManifestTask() {
+        dependsOn(otherProjectProductDependenciesTasks());
+    }
+
+    /**
+     * A lazy collection of tasks that ensure the {@link Jar} task of any project dependencies from
+     * {@link #productDependenciesConfig} is correctly populated with the recommended product dependencies of that
+     * project, if any (specifically, if they apply the {@link RecommendedProductDependenciesPlugin}).
+     */
+    private Provider<FileCollection> otherProjectProductDependenciesTasks() {
+        return productDependenciesConfig.map(productDeps -> {
+            // Using a ConfigurableFileCollection simply because it implements Buildable and provides a convenient API
+            // to wire up task dependencies to it in a lazy way.
+            ConfigurableFileCollection emptyFileCollection = getProject().files();
+            productDeps.getIncoming().getArtifacts().getArtifacts().stream().flatMap(artifact -> {
+                ComponentIdentifier id = artifact.getId().getComponentIdentifier();
+
+                // Depend on the ConfigureProductDependenciesTask, if it exists, which will wire up the jar manifest
+                // with recommended product dependencies.
+                if (id instanceof ProjectComponentIdentifier) {
+                    Project dependencyProject =
+                            getProject().getRootProject().project(((ProjectComponentIdentifier) id).getProjectPath());
+                    return Stream.of(dependencyProject.getTasks().withType(ConfigureProductDependenciesTask.class));
+                }
+                return Stream.empty();
+            }).forEach(emptyFileCollection::builtBy);
```

Comment:
this is a pretty neat trick

Reply:
yeah, I wish this was less obscure, but it's the only way I know how to do it